number_theory/numeric.h

```diff
@@ -92,10 +95,59 @@ auto pow(T base, U exponent) -> decltype(std::pow(base, exponent)) {
   return std::pow(base, exponent);
 }
 
+// Returns the |n|-th root of |x| rounded to integer toward zero.
+// The argument |n| shoud be positive.
+// Throws domain_error exception if the root does not exist.
+// The time complexity is O(log |x|).
+//
+// This overload is only enabled for x < 2^64, because it uses uint64_t
+// to avoid overflows during the computation.
+template <typename T,
+          typename U,
+          std::enable_if_t<std::numeric_limits<T>::digits <= 64, bool> = true>
+T iroot(const T &x, const U &n) {
+  static_assert(
+      std::numeric_limits<T>::is_integer && std::numeric_limits<U>::is_integer,
+      "iroot arguments must be integers.");
+  if (n < 0)
+    throw std::invalid_argument("iroot argument |n| should be positive.");
+  if (n == 0)
```

Comment:
What about combining this condition with the one above, making it `if (n <= 0)`? The comments already say that `n` must be positive. Thus, I think `n <= 0` is invalid.